storage/fanout.go

```diff
@@ -257,53 +257,32 @@ func NewMergeChunkQuerier(primaryQuerier ChunkQuerier, queriers []ChunkQuerier,
 }
 
 // Select returns a set of series that matches the given label matchers.
-func (q *mergeGenericQuerier) Select(sortSeries bool, hints *SelectHints, matchers ...*labels.Matcher) (genericSeriesSet, Warnings, error) {
+func (q *mergeGenericQuerier) Select(sortSeries bool, hints *SelectHints, matchers ...*labels.Matcher) genericSeriesSet {
 	if len(q.queriers) == 1 {
 		return q.queriers[0].Select(sortSeries, hints, matchers...)
 	}
 
-	var (
-		seriesSets = make([]genericSeriesSet, 0, len(q.queriers))
-		warnings   Warnings
-		priErr     error
-	)
 	type queryResult struct {
-		qr          genericQuerier
-		set         genericSeriesSet
-		wrn         Warnings
-		selectError error
+		qr  genericQuerier
+		set genericSeriesSet
 	}
-	queryResultChan := make(chan *queryResult)
-
+	var (
+		seriesSets   = make([]genericSeriesSet, 0, len(q.queriers))
+		queryResults = make(chan *queryResult)
+	)
 	for _, querier := range q.queriers {
 		go func(qr genericQuerier) {
 			// We need to sort for NewMergeSeriesSet to work.
-			set, wrn, err := qr.Select(true, hints, matchers...)
-			queryResultChan <- &queryResult{qr: qr, set: set, wrn: wrn, selectError: err}
+			set := qr.Select(true, hints, matchers...)
+			queryResults <- &queryResult{qr: qr, set: set}
 		}(querier)
 	}
 	for i := 0; i < len(q.queriers); i++ {
-		qryResult := <-queryResultChan
-		q.setQuerierMap[qryResult.set] = qryResult.qr
-		if qryResult.wrn != nil {
-			warnings = append(warnings, qryResult.wrn...)
-		}
-		if qryResult.selectError != nil {
```

Comment:
We might want to ensure correct logic of this in the remote read Select implementation (:

Reply:
Is this addressed?

Reply:
Errors are returned by remote read are now wrapped with `ErrSeriesSet`.

```go
// Select implements storage.Querier and uses the given matchers to read series sets from the Client.
func (q *querier) Select(sortSeries bool, hints *storage.SelectHints, matchers ...*labels.Matcher) storage.SeriesSet {
	query, err := ToQuery(q.mint, q.maxt, matchers, hints)
	if err != nil {
		return storage.ErrSeriesSet(err)
	}

	remoteReadGauge := remoteReadQueries.WithLabelValues(q.client.remoteName, q.client.url.String())
	remoteReadGauge.Inc()
	defer remoteReadGauge.Dec()

	res, err := q.client.Read(q.ctx, query)
	if err != nil {
		return storage.ErrSeriesSet(fmt.Errorf("remote_read: %v", err))
	}

	return FromQueryResult(sortSeries, res)
}
```

Is this what you meant?